sycl/source/detail/platform_impl.cpp

```diff
@@ -140,8 +166,17 @@ static void filterWhiteList(vector_class<RT::PiDevice> &pi_devices) {
                                       info::device::driver_version>::get(dev);
 
     for (const DevDescT &desc : whiteList) {
-      // At least device name is required field to consider the filter so far
-      if (nullptr == desc.devName ||
+      if (nullptr != desc.pltName &&
```

Comment:
Since we get only one platform, can you move this check out of the "foreach devices" loop? And if the check for a platform failed, you can skip the rest.

Reply:
If I do it the filter 
`"PlatformName:{{ptl1}},DeviceName:{{dev1}}|PlatformName{{plt2}},DeviceName{{dev2}}`
will match:
PlatformName = "plt2"
DeviceName = "dev1"
what is not correct.